src/main/java/com/github/elenterius/biomancy/init/AcidInteractions.java

```diff
@@ -157,4 +165,55 @@ else if (livingEntity.tickCount % 10 == 0 && livingEntity.getRandom().nextFloat(
 		}
 	}
 
+	private static final String TIMER_KEY = "biomancy:digestion_timer";
+	private static final float EFFICIENCY = 0.8f;
+	public static void tryDigest(ItemEntity itemEntity) {
+		if (!digestible(itemEntity)) return;
+		CompoundTag data = itemEntity.getPersistentData();
+		if (data.contains(TIMER_KEY) && data.getInt(TIMER_KEY) < 100) {
+			data.putInt(TIMER_KEY,data.getInt(TIMER_KEY)+1);
+			//TODO: Bubble particles? Couldn't get it working for some reason.
+			// Copied the addParticle call from handleEntityInsideAcid and it said no.
+		} else if (data.contains(TIMER_KEY) && data.getInt(TIMER_KEY) >= 100){
+			digestIntoNutrientPasteStacks(itemEntity);
+			itemEntity.getPersistentData().remove(TIMER_KEY);
+		} else {
+			data.putInt(TIMER_KEY,1);
+		}
+	}
+
+	private static void digestIntoNutrientPasteStacks(ItemEntity itemEntity) {
+		if (getDigestionRecipe(itemEntity).isEmpty()) return;
+		DigestingRecipe recipe = getDigestionRecipe(itemEntity).get();
+		BehavioralInventory<?> inv = BehavioralInventory.createServerContents(1,player->false,()->{});
```

Comment:
btw you don't need to use this, you can use the RecipeWrapper class instead

Reply:
I'm not sure what you mean by this, I still need to make an inventory to instantiate a RecipeWrapper?